lib/Re2.fx

```diff
@@ -0,0 +1,828 @@
+    {
+        int fx_status = FX_OK;
+        using namespace re2;   
+        RE2* re_to_apply = static_cast<RE2*>(re->find_r->ptr);
+        fx_re2_findall_process* p_proc = static_cast<fx_re2_findall_process*>(proc->ptr);
+        StringPiece string_n_position(p_proc->cloth.data + pos);
+        fx_result->t0 = RE2::FindAndConsumeN(&string_n_position, *re_to_apply, p_proc->sub_args, p_proc->sub_amount); //TODO: try - catch!
+        if(fx_result->t0)
+        {
+            fx_result->t1 = string_n_position.begin() - p_proc->cloth.data;
+            fx_status = string_pieces_to_arr(p_proc->sub_string_pieces, p_proc->sub_amount, p_proc->cloth.data, &(fx_result->t2));
+        }
+        else
+        {
+            fx_result->t1 = pos;
+        }
+        return fx_status;
+    }
+
+    fun result_as_list(pos: int, lastend: int, reslst: (int, int) [] list) : (int, int)[] list
+    {
+        val (success, newpos, matches) = find_step(proc, pos,re)
+        if(success) 
+        {
+            val (_,newend) = matches[0]
+            val npos = if(newend == lastend) {newpos+1} else {newpos}
+            val newlst = if (newend == lastend) { reslst } else { matches :: reslst }            
+            if(npos < String.length(string_to_match)) 
+            {
+                result_as_list(npos, newend, newlst)
+            }
+            else
+            {
+                List.rev(newlst)
+            }
+        }
+        else
+        {
+            List.rev(reslst)
+        }
+    }
+    val larrist = result_as_list(0, -1, [])
+    val len_i = List.length(larrist)
+    val len_j = if len_i!=0 { size(List.hd(larrist))} else { 0 }
+    val matrix_res: (int,int) [,] = array((len_i, len_j), (0,0))
+    var i = 0
+    for l <- larrist
+    {
+        for j <- 0:len_j
+        {
+            matrix_res[i,j] = l[j]
+        }
+        i += 1 
+    }
+    (len_i !=0, matrix_res)
+}
+
+//                                                     (success, sub-matches[,])
+fun findall_str(string_to_match: string, re: regex_t): (bool   , string     [,])
+{   
+    val (success, starts_ends) = findall(string_to_match, re)
+    val (len_i, len_j) = size(starts_ends)
+    (success, [for i <- 0:len_i for j <- 0:len_j {incise(string_to_match, starts_ends[i,j])}])
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+// Replaces /////////////////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////////////
+
+fun compile_replace_pattern(rewrite: string): replace_pattern_t
+{
+    val digit_reg = compile("\\\\[0-9]") //[TODO] I've tryed to make this val global, but in this case, C++-output became uncompilable.
+    val (has_subs, ranges) = findall(rewrite, digit_reg)
+    val piece_lst = 
+        if(has_subs)
+        {
+            val (len_i, _) = size(ranges)
```

Comment:
the problem with reversed parameters causes by `size()` function cannot be reproduced. Need more information

Reply:
Works fine in latest builds.